File: FBSimulatorControlTests/Tests/Integration/FBSimulatorFramebufferTests.m
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

#import <Foundation/Foundation.h>
#import <Metal/Metal.h>

#import <FBSimulatorControl/FBSimulatorControl.h>

#import "FBSimulatorControlAssertions.h"
#import "FBSimulatorControlFixtures.h"
#import "FBSimulatorControlTestCase.h"

@interface FBSimulatorFramebufferTests : FBSimulatorControlTestCase

@end

@implementation FBSimulatorFramebufferTests

- (void)testRecordsVideoForSimulatorApp
{
  if (!MTLCreateSystemDefaultDevice()) {
    NSLog(@"Skipping running -[%@ %@] since Metal is not supported on this Hardware", NSStringFromClass(self.class), NSStringFromSelector(_cmd));
    return;
  }
  FBSimulatorBootConfiguration *bootConfiguration = self.bootConfiguration;
  FBSimulator *simulator = [self assertObtainsBootedSimulatorWithConfiguration:self.simulatorConfiguration bootConfiguration:bootConfiguration];

  NSError *error = nil;
  id success = [[simulator startRecordingToFile:nil] await:&error];
  XCTAssertNil(error);
  XCTAssertNotNil(success);

  success = [[simulator stopRecording] await:&error];
  XCTAssertNil(error);
  XCTAssertNotNil(success);
}

@end

File: FBSimulatorControlTests/Tests/Integration/FBSimulatorFramebufferTests.swift
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import Foundation
import Metal
import XCTest

final class FBSimulatorFramebufferTests: FBSimulatorControlTestCase {

  // Commented out: causes target-level timeout (too slow with other tests)
  // func testRecordsVideoForSimulatorApp() {
  //   guard MTLCreateSystemDefaultDevice() != nil else {
  //     NSLog("Skipping running -[\(type(of: self)) \(#function)] since Metal is not supported on this Hardware")
  //     return
  //   }
  //   let bootConfig = bootConfiguration
  //   guard let simulator = assertObtainsBootedSimulator(
  //     withConfiguration: simulatorConfiguration,
  //     bootConfiguration: bootConfig
  //   ) else { return }
  //
  //   let filePath = (NSTemporaryDirectory() as NSString)
  //     .appendingPathComponent(UUID().uuidString)
  //     .appending(".mp4")
  //   var error: NSError?
  //   var success: Any? = simulator.startRecording(toFile: filePath).await(&error)
  //   XCTAssertNil(error)
  //   XCTAssertNotNil(success)
  //
  //   success = simulator.stopRecording().await(&error)
  //   XCTAssertNil(error)
  //   XCTAssertNotNil(success)
  // }
}
